app/src/lib/dispatcher/app-store.ts

```diff
@@ -1875,13 +1891,40 @@ export class AppStore {
     return this.refreshGitHubRepositoryInfo(repository)
   }
 
+  private async getAccountForRemoteURL(
+    remote: string
+  ): Promise<IGitAccount | null> {
+    const gitHubRepository = matchGitHubRepository(this.accounts, remote)
+    if (gitHubRepository) {
+      const account = getAccountForEndpoint(
+        this.accounts,
+        gitHubRepository.endpoint
+      )
+      if (account) {
+        return account
+      }
+    }
+
+    const hostname = getGenericHostname(remote)
+    const username = await getGenericUsername(hostname)
```

Comment:
We could... the upside to storing it with `keytar` is that we can more easily tell the user "go delete this entry in Keychain/whatever" than "poke around in the dev tools etc."

I'm not opposed to it, though, if you'd rather we did that.